scripts/cache-inference/cache-inference.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Walrus Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import mdbookOperatorsJson from '../../docs/book/assets/operators.json';
+import { AggregatorData, AggregatorDataVerbose, HeaderValue, Operators } from './types';
+
+const KnownCacheKeys = [
+    "cdn-cache",
+    "cdn-cachedat",
+    "cache-status",
+    "cache-control",
+    "cf-cache-status",
+    "x-cache-status"
+];
+
+type HeaderMatch = {
+    key: string;
+    value: string | null
+};
+type HasCacheOutput = {
+    hasCache: boolean;
+    matches: HeaderMatch[];
+};
+
+function headerKeyContainsCache(headers: Headers): HasCacheOutput {
+    const matches = [...headers.entries()].filter(([key, _]) => {
+        return key.toLowerCase().includes("cache")
+    }).map(([key, value]) => { return { key, value } });
+
+    return {
+        hasCache: matches.length > 0,
+        matches
+    };
+}
+
+function headersHaveCacheHit(matches: HeaderMatch[]): boolean {
+    return matches.some(({ key: _, value }) => {
+        return value?.toLowerCase().includes("hit");
+    });
+}
+
+async function updateAggregatorCacheInfo(
+    aggregators: Record<string, AggregatorData>,
+    blobId: string,
+    threshold: number,
```

Comment:
Nit: If this is always called with `THRESHOLD`, we could just remove the function parameter.